test/integration/concrete/lockup/withdraw-hooks/withdrawHooks.t.sol

```diff
@@ -0,0 +1,260 @@
+        // Create the stream with sender and recipient as contracts.
+        uint256 streamId = createDefaultStream(address(goodRecipient), address(goodSender));
+
+        // Make the sender the caller in this test.
+        changePrank({ msgSender: address(goodSender) });
+
+        // Simulate the passage of time.
+        vm.warp({ timestamp: defaults.WARP_26_PERCENT() });
+
+        // Set the withdraw amount to the default amount.
+        uint128 withdrawAmount = defaults.WITHDRAW_AMOUNT();
+
+        // Expect 1 call to the recipient hook.
+        vm.expectCall({
+            callee: address(goodRecipient),
+            data: abi.encodeCall(
+                ISablierV2Recipient.onLockupStreamWithdrawn,
+                (streamId, address(goodSender), address(goodRecipient), withdrawAmount)
+                ),
+            count: 1
+        });
+
+        // Expect 0 calls to the sender hook.
+        vm.expectCall({
+            callee: address(goodSender),
+            data: abi.encodeCall(
+                ISablierV2Sender.onLockupStreamWithdrawn,
+                (streamId, address(goodSender), address(goodRecipient), withdrawAmount)
+                ),
+            count: 0
+        });
+
+        // Make the withdrawal.
+        lockup.withdraw({ streamId: streamId, to: address(goodRecipient), amount: withdrawAmount });
+    }
+
+    function test_WithdrawHooks_CallerRecipient() external {
+        // Create the stream with sender and recipient as contracts.
+        uint256 streamId = createDefaultStream(address(goodRecipient), address(goodSender));
+
+        // Make the recipient the caller in this test.
+        changePrank({ msgSender: address(goodRecipient) });
+
+        // Simulate the passage of time.
+        vm.warp({ timestamp: defaults.WARP_26_PERCENT() });
+
+        // Set the withdraw amount to the default amount.
+        uint128 withdrawAmount = defaults.WITHDRAW_AMOUNT();
+
+        // Expect 0 calls to the recipient hook.
+        vm.expectCall({
+            callee: address(goodRecipient),
+            data: abi.encodeCall(
+                ISablierV2Recipient.onLockupStreamWithdrawn,
+                (streamId, address(goodRecipient), address(goodRecipient), withdrawAmount)
+                ),
+            count: 0
+        });
+
+        // Expect 1 call to the sender hook.
+        vm.expectCall({
+            callee: address(goodSender),
+            data: abi.encodeCall(
+                ISablierV2Sender.onLockupStreamWithdrawn,
+                (streamId, address(goodRecipient), address(goodRecipient), withdrawAmount)
+                ),
+            count: 1
+        });
+
+        // Make the withdrawal.
+        lockup.withdraw({ streamId: streamId, to: address(goodRecipient), amount: withdrawAmount });
+    }
+
+    modifier givenSenderIsRecipient() {
+        _;
+    }
+
+    function test_SenderHook_CallerUnknown() external givenSenderIsRecipient {
```

Comment:
let's change the name

we should keep the function name first

Reply:
Any suggestions for the name? I thought about it but couldn't come up with names that begins with `test_WithdrawHooks` and fits the description. There were few options like `test_WithdrawHooks_CallerUnknownGiven_SenderIsRecipient` but that would be a very long name.